### PR TITLE
BAC-29 | Create PR to add interface to clean architecture repo

### PR DESCRIPTION
**Pull request by voideus**
- Added auth middleware interface with basic authentication methods.
- Update firebase auth middleware

JIRA
https://wesionary-team.atlassian.net/jira/software/projects/BAC/boards/61?selectedIssue=BAC-29

### CODE REVIEW

File: api/middlewares/auth_middleware.go
Comment (Denes-cilwal) [inline api/middlewares/auth_middleware.go:1]: Why did you remove code base over here ? Please include this 👨🏾

File: api/middlewares/firebase_middleware.go
Comment (Denes-cilwal) [inline api/middlewares/firebase_middleware.go:1]: Same here ! code has been removed try making common function for different service provider !

File: api/routes/user_routes.go
Comment (Denes-cilwal) [inline api/routes/user_routes.go:1]: this is fine 🚀

Comment (Denes-cilwal) [reply api/routes/user_routes.go:1]: Also linter is causing build failed. Add linter setup command before pushing code

File: api/middlewares/auth_middleware.go
Comment (voideus) [reply api/middlewares/auth_middleware.go:1]: Firebase middleware moved to new file (api/middlewares/firebase_middleware.go)

File: api/routes/user_routes.go
Comment (Denes-cilwal) [reply api/routes/user_routes.go:1]: make granular functions , remove multiple middleware for handling auth middleware

**Review by Denes-cilwal — APPROVED** [review]
LGTM 👨🏾
